CODEOWNERS

```diff
@@ -0,0 +1,6 @@
+# These owners will be the default owners for everything in the repo. They
+# will be requested for review when someone opens a pull request.
+# We call those people maintainers as well
+
+*   @displague
```

Comment:
@tinkerbell/site-eng needs to be on here too please.

Reply:
I've been chatting with @mmlb, @kdeng3849, and @rainleander about this process.
Here are some of the points that are relevant to this PR:
* OWNERS.md should be used to define what it means to be an owner
  * This can / should link out to some common tinkerbell org-wide definition of ownership
  * "owner" and "maintainer" should be distinct definitions
    * What should an owner do? What should they not do (when do they defer to the community or proposals)
    * What expectations do we have for maintainers. 
    * Owner may be more of a shepherd, able to call the vet when needed, but otherwise feed and care for the animals, while a maintainer may be more of a vet and breeder. You've heard of clusters as cattle, how about code as sheep? 
* We may benefit from project-specific teams: `@tinkerbell/hegel`, `@tinkerbell/hegel-eng`? (I just made these names up, no idea if they are fitting. maybe `-owners` and `-maintainers` is better?)). The point is that each project may have maintainers that are not actively dealing with the code itself, but provide insight into documentation, branding/marketing, and an SRE perspective on processes and effects that a PR could affect on users.

Reply:
https://github.com/tinkerbell/hegel/pull/29/files#diff-5daf978e60e03b3975b485d5f1b449dcR1-R3 could be the start of OWNERS.md

We can add more details about the community (or better yet, link to it).

I'm ok with making all of these changes later once we get the basic/initial owners in place.

Reply:
Thanks for opening this PR for me @gianarb, so much better than nominating myself in a PR.

Reply:
>     * OWNERS.md should be used to define what it means to be an owner

This should be done before this PR is merged.

>       * This can / should link out to some common tinkerbell org-wide definition of ownership
>       * "owner" and "maintainer" should be distinct definitions

I nominate .github as this central place.

Reply:
I opened https://github.com/tinkerbell/hegel/pull/30 to add site-eng back into this branch and link to a .github/OWNERS.md file (tinkerbell/.github#4).